### PR TITLE
Check the depth of the layer

### PR DESCRIPTION
**Pull request by miaoyq**
Check the depth of the layer to ensure that the depth value is not greater than `maxLayerDepth`

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

**Comment by tonistiigi** [comment]
@miaoyq Can you describe how you found and in what cases it appears. We had a max depth check in `v1.9` as well so not sure how the old data can have more layers.

**Comment by miaoyq** [comment]
There is a max depth check in function `registerWithDescriptor`,  the functions `RegisterByGraphID` and `registerWithDescriptor` are similar, so I think `RegisterByGraphID` should also have the max depth check.

**Review by AkihiroSuda — APPROVED** [review]
I agree this change makes sense, according to the similar code https://github.com/docker/docker/blob/7c62f58b27f98140ff3d39de166bf2e107e84a65/layer/layer_store.go#L265-L268

So LGTM if green

**Comment by thaJeztah** [comment]
@tonistiigi LGTY?

**Comment by stevvooe** [comment]
@thaJeztah Is there an actual bug? Let's not just add code because it looks good. It's possible this check is elsewhere.

**Comment by tonistiigi** [comment]
I have nothing against this change but only thing it changes that it will make migration from Docker `v1.9` fail if for some reason more layers are found on disk. As I said before we had this check in `v1.9` as well so this state should never be reached. So if we haven't had any reports for this in 4 releases and we don't know what cases it could appear I wouldn't change the migration code. If there is a bug that would cause this state and this fix helps to fix it then things would be lot different. Also, this method is not part of the public API so we should only care about the internal migration usage.

**Comment by LK4D4** [comment]
So, looks like @tonistiigi had something against this change. As well as @stevvooe.
I'm going to close this because two maintainers have concerns.
Thanks for your work @miaoyq

**Pull request closed without merging**
